### PR TITLE
Custom Level Text

### PR DESCRIPTION
**Pull request by frinkfromthedrink**
This PR allows users to set a custom text for the users levels, e.g. Rank, Level.

### CODE REVIEW

**Comment by dsevillamartin** [comment]
This PR includes commits from the beta 8 branch that have already been merged. Please go off master.

File: js/src/forum/components/LevelBar.js [hunk 1]
```diff
@@ -7,17 +8,18 @@ export default class LevelBar extends Component {
...
+            expDiscussions = user.discussionCount() * 33;
 
         let expTotal = expComments + expDiscussions,
             expLevel = Math.floor(expTotal / 135),
             expPercent = (100 / 135) * ((expTotal) - (expLevel * 135));
 
         return (
             <div class="PostUser-level" title={expTotal + ' EXP'}>
-                <div class="PostUser-levelNumber">Level {expLevel}</div>
+                <div class="PostUser-levelNumber">{pointsText + ' ' + expLevel}</div>
```
Comment (dsevillamartin) [inline js/src/forum/components/LevelBar.js]: If anything, please do
```suggestion
                <div class="PostUser-levelNumber">{pointsText} {expLevel}</div>
```

Though that won't work with text that wants to be put on the right...

**Review by dsevillamartin — CHANGES_REQUESTED** [review]

File: js/src/forum/components/LevelBar.js (hunk 1 above)
Comment (frinkfromthedrink) [reply js/src/forum/components/LevelBar.js]: In this case, the user could just use `margin-left: 20px` or `margin-right: 20px` to solve that

Comment (dsevillamartin) [reply js/src/forum/components/LevelBar.js]: ...How? It's not a separate <span> tag or anything that the user could select and move around. They'd move the text with the level number.

Comment (frinkfromthedrink) [reply js/src/forum/components/LevelBar.js]: Right.... the text and the current level could be two separate elements wrapped in a div, but that could get more confusing....

Comment (dsevillamartin) [reply js/src/forum/components/LevelBar.js]: Use `<span>`

File: src/Listeners/LoadSettingsFromDatabase.php [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        $events->listen(Serializing::class, [$this, 'prepareApiAttributes']);
+    }
+    /**
+     * @param Serializing $event
+     */
+    public function prepareApiAttributes(Serializing $event)
+    {
+        if ($event->isSerializer(ForumSerializer::class)) {
+            foreach ($this->fieldsToGet as $field) {
+                $value = $this->settings->get($this->packagePrefix.$field);
```
Comment (KyrneDev) [inline src/Listeners/LoadSettingsFromDatabase.php]: What is all this? You're only getting one value so we don't need all this logic.

**Review by KyrneDev — CHANGES_REQUESTED** [review]

File: src/Listeners/LoadSettingsFromDatabase.php (hunk 2 above)
Comment (KyrneDev) [reply src/Listeners/LoadSettingsFromDatabase.php]: Also styleCi

File: js/src/forum/components/LevelBar.js (hunk 1 above)
Comment (frinkfromthedrink) [reply js/src/forum/components/LevelBar.js]: @datitisev would separating the two elements then sticking them in a div work?

Comment (frinkfromthedrink) [reply js/src/forum/components/LevelBar.js]: Ok

File: src/Listeners/LoadSettingsFromDatabase.php (hunk 2 above)
Comment (frinkfromthedrink) [reply src/Listeners/LoadSettingsFromDatabase.php]: @CDK2020 what would you replace that with?

Comment (KyrneDev) [reply src/Listeners/LoadSettingsFromDatabase.php]: Look how it is done here: https://github.com/ReFlar/twofactor/blob/master/src/Listeners/AddApiAttributes.php

Your solution is great when adding tons of things (like in cookie consent) and I'm totally stealing this for the future, but for 1 key, it's bloated.

Comment (frinkfromthedrink) [reply src/Listeners/LoadSettingsFromDatabase.php]: @CDK2020 fixed now

**Comment by frinkfromthedrink** [comment]
@CDK2020 the $key assign is necessary....

**Review by KyrneDev — APPROVED** [review]

**Review by dsevillamartin — APPROVED** [review]

**Comment by frinkfromthedrink** [comment]
@datitisev @CDK2020 ok to merge and release?

**Comment by dsevillamartin** [comment]
Can't release, Shahiem currently is the only one added in Packagist.

**Comment by frinkfromthedrink** [comment]
Won't tagging a new version release it?

**Comment by dsevillamartin** [comment]
No, the webhook isn't set up.

**Comment by frinkfromthedrink** [comment]
Shoot. I'll tag the new version anyway, we can wait or @Shahiem to release it

**Pull request merged**